bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/connection/HubCommunicationThread.java

```diff
@@ -29,17 +32,49 @@
 @NonNullByDefault
 public class HubCommunicationThread extends Thread {
 
+    private enum HubCommunicationThreadState {
+        STARTING(ThingStatus.INITIALIZING, ThingStatusDetail.BRIDGE_UNINITIALIZED, ""),
+        CONNECTED(ThingStatus.ONLINE, ThingStatusDetail.NONE, ""),
+        DISCONNECTED(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/message.bridge.status.failed"),
+        STOPPED(ThingStatus.REMOVING, ThingStatusDetail.BRIDGE_OFFLINE, "");
```

Comment:
This also. See https://www.openhab.org/docs/concepts/things.html#status-transitions.